File: sfn/parallel/SfnParallelResultsAssembler.cs
namespace sfn_ut.sfn.parallel;

public abstract class SfnParallelResultsAssembler<PAYLOAD> : SfnStep {
  public abstract PAYLOAD Execute(PAYLOAD original, List<PAYLOAD> parallelOutputs);

  public SfnParallelResultsAssembler() : base(StepType.ParallelResultsAssembler) {
    // Do nothing
  }
}

File: java-sfn-unit-test/src/main/java/com/sfnut/sfn/parallel/SfnParallelResultsAssembler.java
package com.sfnut.sfn.parallel;

import com.sfnut.sfn.SfnStep;
import com.sfnut.sfn.StepType;
import java.util.List;

public abstract class SfnParallelResultsAssembler<PAYLOAD> extends SfnStep {
    
    public SfnParallelResultsAssembler() {
        super(StepType.PARALLEL_RESULTS_ASSEMBLER);
    }

    public abstract PAYLOAD execute(PAYLOAD original, List<PAYLOAD> parallelOutputs);
}
